grpc/server/admin_test.go

```diff
@@ -29,6 +30,72 @@ func newTestAdminServer(t *testing.T) (*AdminServer, *db.MockQuerier) {
 	}, mockQuerier
 }
 
+func TestAdminServer_IssueAccessToken(t *testing.T) {
+	tests := []struct {
+		name              string
+		token             *commonv1.AccessToken
+		errCode           codes.Code
+		errMsg            string
+		mockQuerierReturn func(*commonv1.AccessToken) (db.IssueAccessTokenRow, error)
+	}{
+		{
+			name:    "issue access token using valid token data request",
+			token:   newAccessTokenBuilder().build(),
+			errCode: codes.OK,
+			errMsg:  "",
+			mockQuerierReturn: func(token *commonv1.AccessToken) (db.IssueAccessTokenRow, error) {
+				tokenID, _ := uuid.Parse(token.Id)
+				return db.IssueAccessTokenRow{
+					ID:                 tokenID,
+					Name:               token.Name,
+					NamespaceSelectors: token.NamespaceSelectors,
+					Scopes:             []string{"admin"},
+					IssuedAt:           types.FromProtoTimestampAsNullTime(token.IssuedAt),
+					ExpiresAt:          types.FromProtoTimestampAsNullTime(token.ExpiresAt),
+				}, nil
+			},
+		},
+		{
+			name:    "db query fails when issuing access token",
+			token:   newAccessTokenBuilder().build(),
+			errCode: codes.Internal,
+			errMsg:  "failed to issue access token",
+			mockQuerierReturn: func(token *commonv1.AccessToken) (db.IssueAccessTokenRow, error) {
+				return db.IssueAccessTokenRow{}, errors.New("Dummy Error")
+			},
+		},
+	}
+
+	server, mockQuerier := newTestAdminServer(t)
+	tokenString := "token"
+	ctx := metadata.NewIncomingContext(context.Background(), metadata.Pairs(auth.MDKeyAuth, "bearer "+tokenString))
```

Comment:
I don't think this is actually needed since we're calling the server's endpoint implementation directly. 

We probably would want to test the auth logic separately in the `github.com/nanzhong/tstr/grpc/auth` package.

(similar comment for the other tests below)

Reply:
you're right. Updated!